### PR TITLE
feat(codemirror): Add highlightError, highlightWarning and highlightInfo

### PR DESCRIPTION
**Pull request by jbrundage**
Signed-off-by: Jaman Brundage <jbrundage372@gmail.com>

## Checklist:
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit message includes a "fixes" reference if appropriate.
  - [x] The commit is signed.
- [x] The change has been fully tested:
  - [ ] I have viewed all related gallery items
  - [x] I have viewed all related dermatology items
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised new issues to address them separately

### CODE REVIEW

**Comment by jbrundage** [comment]
image

**Comment by GordonSmith** [comment]
@jbrundage - did you see:  https://codemirror.net/demo/lint.html

**Comment by jbrundage** [comment]
@GordonSmith I've amended this PR to use their encoded background image approach

File: packages/codemirror/src/Editor.ts [hunk 1]
```diff
@@ -16,7 +16,9 @@ export class Editor extends HTMLWidget {
     options(): any {
         return {
             lineNumbers: true,
-            tabSize: 2
+            tabSize: 2,
+            gutters: ["CodeMirror-lint-markers"],
+            lint: true
```
Comment (GordonSmith) [inline packages/codemirror/src/Editor.ts]: I think the show "lint" should be optional "showHighlightGutter"?

File: packages/codemirror/src/Editor.ts [hunk 2]
```diff
@@ -82,6 +91,7 @@ export class Editor extends HTMLWidget {
     update(domNode, Element) {
         super.update(domNode, Element);
         this._codemirror.setSize(this.width() - 2, this.height() - 2);
+        this._codemirror
```
Comment (GordonSmith) [inline packages/codemirror/src/Editor.ts]: Typo?

File: packages/codemirror/src/Editor.ts [hunk 3]
```diff
@@ -38,13 +40,20 @@ export class Editor extends HTMLWidget {
...
+            this._markedText.push(this._codemirror.markText(start, end, {className}));
         }
         return this;
     }
 
+    highlightError(start: IPosition | number, end: IPosition | number): this {
+        this.highlight(start, end, "cm-marked-error");
+        return this;
+    }
+
```
Comment (GordonSmith) [inline packages/codemirror/src/Editor.ts]: Need a way to highlight warnings / info and errors?

File: packages/codemirror/src/Editor.ts (hunk 1 above)
Comment (jbrundage) [reply packages/codemirror/src/Editor.ts]: I don't recall why I added "lint" in this pull request. Did we discuss it? It seems like it's beyond the scope of this change and would need this addon to be loaded (in codemirror-shim?):
https://codemirror.net/addon/lint/lint.js

As well as these addons (as needed?):
https://codemirror.net/addon/lint/javascript-lint.js
https://codemirror.net/addon/lint/json-lint.js
https://codemirror.net/addon/lint/css-lint.js
https://codemirror.net/addon/lint/html-lint.js

**Pull request merged**